xdsl/irdl.py

```diff
@@ -473,13 +488,33 @@ def from_pyrdl(pyrdl_def: type[_OpT]) -> OpDef:
                 continue
             args = get_args(field_type)
 
-            if isinstance(args[-1], OperandDef):
-                op_def.operands.append((field_name, args[-1]))
-            elif isinstance(args[-1], ResultDef):
-                op_def.results.append((field_name, args[-1]))
+            fail = False
+            if len(args) == 3:
```

Comment:
Actually, we shouldn't reason about the size of the arguments of `Annotated`.
For instance, `Annotated[A, Annotated[B, C]]` is simplified by Python to `Annotated[A, B, C]`.